### ISSUE TITLE
Mobile sync taking up to 10 seconds per sync

### ISSUE THREAD

**Issue by dianetoggl** [toggl/mobileapp#7176]
## ⛑️ User Support

Sync is taking up to 10 seconds for certain users. Android and iOS are both affected. This includes when you open the app, when you start a time entry, when you stop a time entry and when you pull down to manually sync.

## About the user

[Link to Google Drive sheet](https://docs.google.com/spreadsheets/d/1wkAWV_NYc_7IAGiWkR30y9Tg4GwRGvxvF2EaQTz6O_s/edit?usp=sharing)

## Expected behaviour

Sync would be a lot quicker 

## To Reproduce

[Please see linked video](https://drive.google.com/open?id=1pQVtm09PTiO5mcHmTID-ZzE3ppKI60md)
(This is my own video from my personal phone - no user data apart from my own)

**Comment by amulware** [comment]
Categorising as bug for the impact this has on UX. Push sync should take less than a second in most cases.

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/mentions/conversations/26269783351)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/unassigned/conversations/26257737755)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/3540765/conversations/26310748488)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/3540765/conversations/26389738393)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/unassigned/conversations/26500269414)

**Comment by dshr** [comment]
I couldn't find anything sensible — syncs would sometimes work, and sometimes would wait for a response for a while. Still, it wasn't that we were getting hit with the 1 minute wait time, but something else 🤷‍♂

**Comment by dianetoggl** [comment]
@dshr I don't mind jumping on at a time that works for you and going through this with you - I have the slow sync consistently. It has taken 10 second each and every time I have to sync a time entry or load the app 😊

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/unassigned/conversations/26639978498)

**Comment by dianetoggl** [comment]
[Case [](reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/3540765/conversations/26614819449))

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/3540765/conversations/26103459681)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/unassigned/conversations/26808497373)

**Comment by nicktoggl** [comment]
[Case Reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/2097915/conversations/26671058931)

**Comment by nidarasheed** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/534741/conversations/26775849335)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/unassigned/conversations/26850399198)

**Comment by dianetoggl** [comment]
[Case reference](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/unassigned/conversations/26913708440)

**Comment by pdrj** [comment]
As with #7238 the problem started happening in v2.10. I've used v2.9 apk and the sync is way quicker than in v2.10.

I'll do a bisect now to pinpoint the bad commit.

**Comment by pdrj** [comment]
Caused by #6826 😿

**Comment by pdrj** [comment]
I've timed how long the api takes to respond with #6826 applied:

```
2020-05-07 11:24:55.825 31915-32072/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/workspaces took 385
2020-05-07 11:24:56.404 31915-31997/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/features took 963
2020-05-07 11:24:59.485 31915-32098/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 4041
2020-05-07 11:25:00.977 31915-32098/? I/mono-stdout: xxaa 11:25 AM GOT reponse https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 5526
2020-05-07 11:25:01.683 31915-32072/? I/mono-stdout: xxaa 11:25 AM GOT reponse https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 6236
2020-05-07 11:25:02.438 31915-32069/? I/mono-stdout: xxaa 11:25 AM GOT reponse https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 6989
2020-05-07 11:25:03.949 31915-31986/? I/mono-stdout: xxaa 11:25 AM GOT reponse https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 8495
```

```
2020-05-07 11:24:27.022 31915-32098/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/workspaces took 228
2020-05-07 11:24:29.301 31915-31986/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/features took 2505
2020-05-07 11:24:30.827 31915-32070/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 4027
2020-05-07 11:24:31.594 31915-32071/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 4795
2020-05-07 11:24:32.352 31915-31987/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 5516
2020-05-07 11:24:33.875 31915-31987/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 7037
2020-05-07 11:24:34.620 31915-32070/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 7786
```

```
2020-05-07 11:24:10.462 31915-32011/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/workspaces took 264
2020-05-07 11:24:13.515 31915-32098/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/features took 3315
2020-05-07 11:24:14.973 31915-32072/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 4736
2020-05-07 11:24:15.760 31915-31986/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 5521
2020-05-07 11:24:16.512 31915-32011/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 6272
2020-05-07 11:24:17.261 31915-31997/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 7019
2020-05-07 11:24:18.791 31915-31987/? I/mono-stdout: xxaa 11:24 AM GOT reponse https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 8557
```

and with one commit prior:
```
2020-05-07 11:32:18.800 482-814/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/workspaces took 367
2020-05-07 11:32:18.804 482-803/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/features took 368
2020-05-07 11:32:19.814 482-813/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/clients?since=1585741153
2020-05-07 11:32:19.822 482-808/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/tags?since=1584969675
2020-05-07 11:32:20.069 482-802/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True
2020-05-07 11:32:20.077 482-1316/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 262
2020-05-07 11:32:20.082 482-808/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 259
2020-05-07 11:32:20.084 482-815/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/tasks?since=1586861924
2020-05-07 11:32:20.088 482-808/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867
2020-05-07 11:32:20.305 482-1314/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 236
2020-05-07 11:32:20.318 482-803/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 230
2020-05-07 11:32:20.318 482-1316/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 234
```

```

2020-05-07 11:32:49.231 482-482/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/workspaces
2020-05-07 11:32:49.235 482-482/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/features
2020-05-07 11:32:49.477 482-812/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/workspaces took 245
2020-05-07 11:32:49.479 482-803/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/features took 243
2020-05-07 11:32:50.484 482-1314/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/tags?since=1584969675
2020-05-07 11:32:50.486 482-809/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/clients?since=1585741153
2020-05-07 11:32:50.689 482-802/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 203
2020-05-07 11:32:50.700 482-814/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 216
2020-05-07 11:32:51.704 482-814/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/tasks?since=1586861924
2020-05-07 11:32:51.707 482-809/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867
2020-05-07 11:32:51.717 482-815/? I/mono-stdout: xxaa 11:32 AM send request to https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True
2020-05-07 11:32:51.938 482-808/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 235
2020-05-07 11:32:51.956 482-814/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 248
2020-05-07 11:32:51.957 482-1314/? I/mono-stdout: xxaa 11:32 AM GOT reponse https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 240
```

The difference is huge.

The logging happens in `SendRequest` in `BaseApi.cs`

**Comment by pdrj** [comment]
Further info:
```
2020-05-07 12:28:58.828 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/workspaces
2020-05-07 12:28:58.828 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.828 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/workspaces
2020-05-07 12:28:58.828 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/workspaces - wait for slot
2020-05-07 12:28:58.828 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/workspaces - got slot
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me - wait for slot
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/features
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/features
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/features - wait for slot
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/preferences
2020-05-07 12:28:58.831 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/preferences - wait for slot
2020-05-07 12:28:58.870 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/tags?since=1584969675
2020-05-07 12:28:58.870 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.870 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tags?since=1584969675
2020-05-07 12:28:58.870 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tags?since=1584969675 - wait for slot
2020-05-07 12:28:58.872 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/clients?since=1585741153
2020-05-07 12:28:58.872 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.872 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/clients?since=1585741153
2020-05-07 12:28:58.872 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/clients?since=1585741153 - wait for slot
2020-05-07 12:28:58.874 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True
2020-05-07 12:28:58.874 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.874 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True
2020-05-07 12:28:58.874 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True - wait for slot
2020-05-07 12:28:58.876 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867
2020-05-07 12:28:58.876 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.876 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867
2020-05-07 12:28:58.876 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 - wait for slot
2020-05-07 12:28:58.877 8270-8270/? I/mono-stdout: xxaa 12:28 PM send request to https://mobile.toggl.space/api/v9/me/tasks?since=1586861924
2020-05-07 12:28:58.877 8270-8270/? I/mono-stdout: xxaa header Authorization -> YzM2NzdkZmQ3YmI2M2E4OTkxNTEyM2EwNzFhNTBhMTU6YXBpX3Rva2Vu
2020-05-07 12:28:58.877 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tasks?since=1586861924
2020-05-07 12:28:58.877 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 - wait for slot
2020-05-07 12:28:59.077 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/workspaces - send returned
2020-05-07 12:28:59.077 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/workspaces took 248
2020-05-07 12:28:59.077 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:28:59.079 8270-8383/? I/mono-stdout: xxaa 12:28 PM GOT reponse https://mobile.toggl.space/api/v9/me/workspaces took 251
2020-05-07 12:28:59.584 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me - got slot
2020-05-07 12:28:59.776 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me - send returned
2020-05-07 12:28:59.776 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me took 945
2020-05-07 12:28:59.776 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:00.339 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/features - got slot
2020-05-07 12:29:00.577 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/features - send returned
2020-05-07 12:29:00.577 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/features took 1745
2020-05-07 12:29:00.577 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:00.580 8270-8384/? I/mono-stdout: xxaa 12:29 PM GOT reponse https://mobile.toggl.space/api/v9/me/features took 1748
2020-05-07 12:29:01.094 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/preferences - got slot
2020-05-07 12:29:01.303 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/preferences - send returned
2020-05-07 12:29:01.304 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/preferences took 2472
2020-05-07 12:29:01.304 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:02.637 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True - got slot
2020-05-07 12:29:02.861 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True - send returned
2020-05-07 12:29:02.861 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 3987
2020-05-07 12:29:02.861 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:02.863 8270-8424/? I/mono-stdout: xxaa 12:29 PM GOT reponse https://mobile.toggl.space/api/v9/me/projects?since=1586861924&include_archived=True took 3989
2020-05-07 12:29:03.394 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/clients?since=1585741153 - got slot
2020-05-07 12:29:03.615 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/clients?since=1585741153 - send returned
2020-05-07 12:29:03.616 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 4743
2020-05-07 12:29:03.616 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:03.617 8270-8392/? I/mono-stdout: xxaa 12:29 PM GOT reponse https://mobile.toggl.space/api/v9/me/clients?since=1585741153 took 4744
2020-05-07 12:29:04.150 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 - got slot
2020-05-07 12:29:04.407 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 - send returned
2020-05-07 12:29:04.407 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 5531
2020-05-07 12:29:04.407 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:04.409 8270-8415/? I/mono-stdout: xxaa 12:29 PM GOT reponse https://mobile.toggl.space/api/v9/me/time_entries?since=1588588867 took 5533
2020-05-07 12:29:05.659 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 - got slot
2020-05-07 12:29:05.869 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 - send returned
2020-05-07 12:29:05.869 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 6991
2020-05-07 12:29:05.869 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:05.870 8270-8416/? I/mono-stdout: xxaa 12:29 PM GOT reponse https://mobile.toggl.space/api/v9/me/tasks?since=1586861924 took 6993
2020-05-07 12:29:06.417 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tags?since=1584969675 - got slot
2020-05-07 12:29:06.680 8270-8270/? I/mono-stdout: xxaa RateLimitingAwareApiClient send https://mobile.toggl.space/api/v9/me/tags?since=1584969675 - send returned
2020-05-07 12:29:06.681 8270-8270/? I/mono-stdout: xxaa AdaptingRateLimitingAwareApiClient GOT https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 7810
2020-05-07 12:29:06.681 8270-8270/? I/mono-stdout: xxaa change size leaky bucket 80 slot seconds 0 maxslots 1 rem slots 0
2020-05-07 12:29:06.682 8270-8423/? I/mono-stdout: xxaa 12:29 PM GOT reponse https://mobile.toggl.space/api/v9/me/tags?since=1584969675 took 7811
```

**Comment by pdrj** [comment]
This is waiting for some descisions: https://toggl.slack.com/archives/C02APMYTM/p1588934950167400

TLDR is that we're mostly limited by backend requirements to not send too many requests at once, which makes the sync process slow.
